src/zenml/cli/stack.py

```diff
@@ -681,3 +714,77 @@ def down_stack(force: bool = False) -> None:
             f"Suspending resources for active stack '{stack_.name}'."
         )
         stack_.suspend()
+
+
+@stack.command("export")
+@click.argument("stack_name", type=str, required=True)
+@click.argument("filename", type=str, required=True)
+def export_stack(stack_name: str, filename: str) -> None:
+    """Export a stack to YAML."""
+
+    # Get configuration of given stack
+    # TODO: code duplicate with describe()
+    repo = Repository()
+    stack_configurations = repo.stack_configurations
+    if len(stack_configurations) == 0:
+        cli_utils.warning("No stacks registered!")
+        return
+    try:
+        stack_configuration = stack_configurations[stack_name]
+    except KeyError:
+        cli_utils.error(f"Stack '{stack_name}' does not exist.")
+        return
+
+    # create a dict of all components in the specified stack
+    component_data = {}
+    for component_type, component_name in stack_configuration.items():
+        components = repo.get_stack_components(component_type)
```

Comment:
I think you could shortcut some of the code here by using `repo.get_stack_component(component_type, name=component_name)`

Reply:
Makes sense, that's much cleaner.